### PR TITLE
Feature/connect burger with mobile menu

### LINKED ISSUE #7: Mobile Navigation still not connected to o-header
The burger button of `m-main-navigation` needs to be able to talk to `m-main-navigation-mobile`

Related: #26 #31

### LINKED ISSUE #35: Keep existing css classes
If a custom component has some classes attached to it, then they should be kept and not thrown away.

**Edit:**

```html
<axa-custom class="any-classes-here"></axa-custom>
```

```js
Custom extends HTMLElement {
  connectedCallback() {
    // remove all classes, in above case: .any-classes-here
    this.className = 'a-custom';
  }
}

window.customElements.define('axa-custom', Custom);
```

Results in:
```html
<axa-custom class="a-custom"></axa-custom>
```

### PR DESCRIPTION
**Pull request by AndyOGo**
fixes #7 
fixes #35 

suffers from #31 

# Todo

- [x] hide/show `m-main-navigation-mobile` upon `m-main-navigation__burger`'s click
- [x] add drop shadow
- [x] lock document scrolling
- [x] proper solution does all of above only within the boundaries of a given context

### CODE REVIEW

File: src/components/m-main-navigation-mobile/index.scss [hunk 1]
```diff
@@ -18,14 +38,36 @@
...
+  position: absolute;
+  top: 100%;
+  right: 0;
+  z-index: $main-navigation-mobile__nav-z-index;
+
+  height: 100%;
+  margin-top: -1px;
+  overflow-x: hidden;
+  overflow-y: scroll;
+  -webkit-overflow-scrolling: touch;
```
Comment (LucaMele) [inline src/components/m-main-navigation-mobile/index.scss]: wir brauchen noch den autoprefixer

File: src/components/m-main-navigation/js/burger.js [hunk 2]
```diff
@@ -58,28 +95,48 @@ class Burger {
...
     if (!this.isOpen) {
       return;
     }
 
     this.isOpen = false;
 
     remove(this.burger, this.options.burgerState);
+
+    if (!e && this._contextNode) {
+      console.log('<<<<<<<<< publish close');
```
Comment (LucaMele) [inline src/components/m-main-navigation/js/burger.js]: do we need this logs?

File: src/components/m-main-navigation-mobile/js/mobile-navigation.js [hunk 3]
```diff
@@ -18,33 +22,90 @@ class MobileNavigation {
...
+      this.unSubscribeOpen();
+    }
+
+    if (this.unSubscribeClose) {
+      this.unSubscribeClose();
+    }
+  }
+
+  open() {
+    console.log('open mobile menu');
```
Comment (LucaMele) [inline src/components/m-main-navigation-mobile/js/mobile-navigation.js]: console log?

Comment (AndyOGo) [reply src/components/m-main-navigation-mobile/js/mobile-navigation.js]: WIP

File: src/components/m-main-navigation/js/burger.js (hunk 2 above)
Comment (AndyOGo) [reply src/components/m-main-navigation/js/burger.js]: WIP, will be removed before merge

File: src/components/m-main-navigation-mobile/index.scss (hunk 1 above)
Comment (AndyOGo) [reply src/components/m-main-navigation-mobile/index.scss]: Absolutely, but I guess `-webkit-overflow-scrolling` is webkit only

**Comment by AndyOGo** [comment]
Buuuug

>index.js:1981 Uncaught TypeError: this.interaction.destroy is not a function
    at HTMLElement.disconnectedCallback (index.js:1981)
    at HTMLElement.render (index.js:1119)
    at HTMLElement.connectedCallback (index.js:1051)
    at HTMLElement.connectedCallback (index.js:2302)
    at HTMLDocument.<anonymous> (index.js:2329)

**Comment by AndyOGo** [comment]
@LucaMele 
Please review again.
I added fixes for found issues, regarding:

- scroll to top (very hacky though)
- height of sub menu
- restoring scroll position within sub menu
- always start with initial state

**Pull request merged**